File: app/src/main/java/chen/vike/c680/ALiPay/ThreadPoolManager.java
package chen.vike.c680.ALiPay;

import com.blankj.utilcode.util.LogUtils;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

public class ThreadPoolManager {
    private ExecutorService service;

    private ThreadPoolManager() {
        int num = Runtime.getRuntime().availableProcessors();
        service = Executors.newFixedThreadPool(num * 2);
    }

    private static final ThreadPoolManager manager = new ThreadPoolManager();

    public static ThreadPoolManager getInstance() {
        return manager;
    }

    public void addTask(Runnable runnable) {
        String TAG = "ThreadPoolManager";
        LogUtils.v(TAG, "加载任务");
        service.execute(runnable);
    }
}

File: app/src/main/java/chen/vike/c680/tools/ThreadPoolManager.kt
package chen.vike.c680.tools

import android.util.Log

import java.util.concurrent.ExecutorService
import java.util.concurrent.Executors


class ThreadPoolManager private constructor() {
    private val service: ExecutorService

    init {
        val num = Runtime.getRuntime().availableProcessors()
        service = Executors.newFixedThreadPool(num * 2)
    }

    fun addTask(runnable: Runnable) {
        val tag = "ThreadPoolManager"
        Log.v(tag, "加载任务")
        service.execute(runnable)
    }

    fun shutDown(){
        if (!service.isShutdown || !service.isTerminated){
            service.shutdown()
        }
    }

    companion object {

        val instance = ThreadPoolManager()
    }
}
